direct/transport/tpm.go

```diff
@@ -10,20 +10,77 @@ import (
 // TPM represents a logical connection to a TPM.
 type TPM interface {
 	Send(input []byte) ([]byte, error)
-	Close() error
 }
 
-// LocalTPM represents a connection to the local TPM.
-type LocalTPM struct {
+// TPMCloser represents a logical connection to a TPM and you can close it.
+type TPMCloser interface {
+	TPM
+	io.Closer
+}
+
+// The wrappedRW represents a struct that wraps an io.ReadWriter
+// to function like a transport.TPM.
+type wrappedRW struct {
+	transport io.ReadWriter
+}
+
+// The wrappedRWC represents a struct that wraps an io.ReadWriteCloser
+// to function like a transport.TPM with the close function.
+type wrappedRWC struct {
 	transport io.ReadWriteCloser
 }
 
+// The wrappedTPM represents a struct that wraps a transport.TPM 
+// to function like a io.ReadWriter
```

Comment:
nit: Maybe rephrase, like:

wrappedTPM represents a struct that wraps a transport.TPM's underlying transport to use with legacy code that expects an io.ReadWriter

Reply:
100% agreed, I was having trouble describing it myself, thank you!